MantidQt/CustomInterfaces/test/ReflCommandsTest.h

```diff
@@ -0,0 +1,321 @@
+#ifndef MANTID_CUSTOMINTERFACES_REFLCOMMANDSTEST_H
+#define MANTID_CUSTOMINTERFACES_REFLCOMMANDSTEST_H
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidQtCustomInterfaces/Reflectometry/IReflTablePresenter.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflAppendRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflClearSelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflCopySelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflCutSelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflDeleteRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflExpandCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflExportTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflGroupRowsCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflImportTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflNewTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflOpenTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflOptionsCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPasteSelectedCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPlotGroupCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPlotRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflPrependRowCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflProcessCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflSaveTableAsCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflSaveTableCommand.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflTableViewPresenter.h"
+
+using namespace MantidQt::CustomInterfaces;
+using namespace Mantid::API;
+using namespace testing;
+
+class MockIReflTablePresenter : public IReflTablePresenter {
+
+public:
+  MOCK_METHOD1(notify, void(IReflTablePresenter::Flag));
+
+private:
+  // Calls we don't care about
+  const std::map<std::string, QVariant> &options() const { return m_options; };
```

Comment:
If you don't care about calls. I think the thing to do is to create a constructor for the Mock and in it use `ON_CALL' to define what happens when the function is called, namely to define what you're going to return because that's a problem even when you're using `NiceMock`. That way I think you can still override the `EXPECT_CALL` you want using the same Mock type. Hope this makes sense.